File: packages/python/examples/basic.py
"""Basic FlowTTS Example."""

import os

from flow_tts import FlowTTS

try:
    from dotenv import load_dotenv

    load_dotenv()
except ImportError:
    pass


def main() -> None:
    """Run basic TTS example."""
    print("🎤 FlowTTS Basic Example\n")

    # Initialize client
    client = FlowTTS(
        {
            "secret_id": os.getenv("TX_SECRET_ID", ""),
            "secret_key": os.getenv("TX_SECRET_KEY", ""),
            "sdk_app_id": int(os.getenv("TRTC_SDK_APP_ID", "0")),
        }
    )

    # 1. Synthesize Chinese text
    print("1️⃣ Synthesizing Chinese text...")
    try:
        response = client.synthesize(
            {"text": "你好，世界！这是一个测试。", "format": "wav"}
        )

        with open("output-chinese.wav", "wb") as f:
            f.write(response["audio"])

        print(f"✅ Success! Generated {len(response['audio'])} bytes")
        print(
            f"   Language: {response.get('detected_language')} (auto-detected: {response.get('auto_detected')})"
        )
        print(f"   Request ID: {response['request_id']}")
        print("💾 Saved to output-chinese.wav\n")
    except Exception as e:
        print(f"❌ Error: {e}\n")

    # 2. Synthesize English text
    print("2️⃣ Synthesizing English text...")
    try:
        response = client.synthesize(
            {"text": "Hello, world! This is a test.", "format": "wav"}
        )

        with open("output-english.wav", "wb") as f:
            f.write(response["audio"])

        print(f"✅ Success! Generated {len(response['audio'])} bytes")
        print(f"   Language: {response.get('detected_language')}")
        print("💾 Saved to output-english.wav\n")
    except Exception as e:
        print(f"❌ Error: {e}\n")

    # 3. Get available voices
    print("3️⃣ Listing available voices...")
    voices = client.get_voices(include_extended=False)
    print(f"Found {len(voices['preset'])} voices:")
    for voice in voices["preset"][:5]:
        print(
            f"   - {voice['id']}: {voice['name']} ({voice.get('language', 'unknown')})"
        )

    # 4. Search voices
    print("\n4️⃣ Searching for '温柔' voices...")
    gentle_voices = client.search_voices("温柔")
    print(f"Found {len(gentle_voices)} gentle voices:")
    for voice in gentle_voices[:3]:
        print(f"   - {voice['id']}: {voice['name']}")


if __name__ == "__main__":
    main()

File: packages/node/examples/basic.ts
/**
 * Basic TTS Example
 * Demonstrates simple text-to-speech synthesis
 */

import 'dotenv/config';
import { FlowTTS } from '../src/index.js';
import { writeFileSync } from 'fs';

async function main() {
  // Initialize client with credentials
  const client = new FlowTTS({
    secretId: process.env.TX_SECRET_ID!,
    secretKey: process.env.TX_SECRET_KEY!,
    sdkAppId: parseInt(process.env.TRTC_SDK_APP_ID!),
    region: 'ap-beijing' // Optional, defaults to ap-beijing
  });

  console.log('🎤 FlowTTS Basic Example\n');

  // Example 1: Synthesize Chinese text
  console.log('1️⃣ Synthesizing Chinese text...');
  const response1 = await client.synthesize({
    text: '你好，世界！欢迎使用 FlowTTS SDK。',
    voice: 'v-female-R2s4N9qJ', // 温柔姐姐
    format: 'wav'
  });

  console.log(`✅ Success! Generated ${response1.audio.length} bytes`);
  console.log(`   Language: ${response1.detectedLanguage} (auto-detected: ${response1.autoDetected})`);
  console.log(`   Request ID: ${response1.requestId}\n`);

  // Save to file
  writeFileSync('output-chinese.wav', response1.audio);
  console.log('💾 Saved to output-chinese.wav\n');

  // Example 2: Synthesize English text
  console.log('2️⃣ Synthesizing English text...');
  const response2 = await client.synthesize({
    text: 'Hello, world! Welcome to FlowTTS SDK.',
    language: 'en', // Explicitly specify language
    format: 'wav'
  });

  console.log(`✅ Success! Generated ${response2.audio.length} bytes`);
  console.log(`   Language: ${response2.detectedLanguage || 'en'}\n`);

  writeFileSync('output-english.wav', response2.audio);
  console.log('💾 Saved to output-english.wav\n');

  // Example 3: List available voices
  console.log('3️⃣ Listing available voices...');
  const { preset } = client.getVoices(false); // Standard voices only
  console.log(`Found ${preset.length} voices:`);
  preset.slice(0, 5).forEach(voice => {
    console.log(`   - ${voice.id}: ${voice.name} (${voice.language})`);
  });
  console.log(`   ... and ${preset.length - 5} more\n`);

  // Example 4: Search voices
  console.log('4️⃣ Searching for "温柔" voices...');
  const searchResults = client.searchVoices('温柔');
  console.log(`Found ${searchResults.length} matching voices:`);
  searchResults.forEach(voice => {
    console.log(`   - ${voice.id}: ${voice.name}`);
  });

  console.log('\n✨ All examples completed successfully!');
}

main().catch(console.error);
